### PR TITLE
Box view

### PR DESCRIPTION
**Pull request by vishkaischibsted**
Add `Box` component.

Screenshot 2024-02-09 at 13 27 55

### CODE REVIEW

File: Sources/Box/Box.swift [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+            hasher.combine(title)
+        }
+
+        public init(
+            style: Warp.BoxStyle,
+            title: String?,
+            subtitle: String,
+            link: ButtonConstructor? = nil,
+            button: ButtonConstructor? = nil,
+            secondaryButton: ButtonConstructor? = nil,
```
Comment (mohsen-schibsted) [inline Sources/Box/Box.swift:49]: box only has one button, and maybe add the missing self so it looks the same

Comment (vishkaischibsted) [reply Sources/Box/Box.swift]: Done.

File: Sources/Box/Box.swift [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+                informationView
+                    .padding(.leading, 16)
+
+                Spacer()
+            }
+        }
+
+        @ViewBuilder
+        private var toolTipIconView: some View {
+            if true {
```
Comment (mohsen-schibsted) [inline Sources/Box/Box.swift:120]: if there is no title this should be hidden too.

Comment (vishkaischibsted) [reply Sources/Box/Box.swift:120]: Great catch, I will update it.
I will make it injectable.

File: Sources/Box/Box.swift [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+                buttonsView
+                    .padding(.top, 4)
+                    .padding(.bottom, 8)
+            }
+        }
+
+        @ViewBuilder
+        private var titleView: some View {
+            if let title {
+                Text(title)
```
Comment (mohsen-schibsted) [inline Sources/Box/Box.swift:151]: why not use our own Text(title, style: .title3)?

Comment (vishkaischibsted) [reply Sources/Box/Box.swift]: Done, thanks.

File: Sources/Text/Text.swift [hunk 4]
```diff
@@ -30,7 +30,6 @@ extension Warp {
         public var body: some View {
             SwiftUI.Text(text)
                 .font(from: style.asTypography)
-                .foregroundColor(colorProvider.labelText)
```
Comment (mohsen-schibsted) [inline Sources/Text/Text.swift:33]: if you remove this then other places that uses our own Warp.Text suddenly lost their colors and became .black, right?

Comment (vishkaischibsted) [reply Sources/Text/Text.swift]: Good point, I will revert this.

**Review by mohsen-schibsted — APPROVED** [review]

**Pull request merged**